File: app/src/main/java/com/worldventures/dreamtrips/modules/profile/model/ReloadFeedModel.java
package com.worldventures.dreamtrips.modules.profile.model;

import android.os.Parcel;
import android.os.Parcelable;

public class ReloadFeedModel implements Parcelable {
   boolean visible;

   public boolean isVisible() {
      return visible;
   }

   public void setVisible(boolean visible) {
      this.visible = visible;
   }


   @Override
   public int describeContents() {
      return 0;
   }

   @Override
   public void writeToParcel(Parcel dest, int flags) {
      dest.writeByte(visible ? (byte) 1 : (byte) 0);
   }

   public ReloadFeedModel() {
   }

   protected ReloadFeedModel(Parcel in) {
      this.visible = in.readByte() != 0;
   }

   public static final Creator<ReloadFeedModel> CREATOR = new Creator<ReloadFeedModel>() {
      public ReloadFeedModel createFromParcel(Parcel source) {
         return new ReloadFeedModel(source);
      }

      public ReloadFeedModel[] newArray(int size) {
         return new ReloadFeedModel[size];
      }
   };
}

File: app/src/main/java/com/worldventures/dreamtrips/social/service/profile/model/ReloadFeedModel.kt
package com.worldventures.dreamtrips.social.service.profile.model

import android.os.Parcel
import android.os.Parcelable

class ReloadFeedModel() : Parcelable {

   var isVisible: Boolean = false

   protected constructor(parcel: Parcel) : this() {
      this.isVisible = parcel.readByte().toInt() != 0
   }

   override fun describeContents() = 0

   override fun writeToParcel(dest: Parcel, flags: Int) {
      dest.writeByte(if (isVisible) 1 else 0)
   }

   companion object {
      val CREATOR: Parcelable.Creator<ReloadFeedModel> = object : Parcelable.Creator<ReloadFeedModel> {
         override fun createFromParcel(source: Parcel) = ReloadFeedModel(source)
         override fun newArray(size: Int): Array<ReloadFeedModel?> = arrayOfNulls(size)
      }
   }
}
